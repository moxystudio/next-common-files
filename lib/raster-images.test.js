'use strict';

const { withRasterImages } = require('../index.js');

const options = {
    dev: true,
    isServer: true,
};

const mockFunction = jest.fn();

let config;

beforeEach(() => {
    config = {
        module: {
            rules: [],
        },
    };

    jest.clearAllMocks();
});

it('should match snapshot', () => {
    const arrayOfImagesRules = [];

    arrayOfImagesRules.push(withRasterImages()().webpack(config, options).module.rules);

    expect(arrayOfImagesRules).toMatchSnapshot();
});

it('should return a single rule', () => {
    const arrayOfImagesRules = [];

    arrayOfImagesRules.push(withRasterImages()().webpack(config, options).module.rules);

    expect(arrayOfImagesRules).toHaveLength(1);
});

it('should return a function if next config is a function', () => {
    const nextConfig = {
        webpack: mockFunction,
    };

    withRasterImages()(nextConfig).webpack(config, options);

    expect(nextConfig.webpack.mock.calls).toHaveLength(1);
});
