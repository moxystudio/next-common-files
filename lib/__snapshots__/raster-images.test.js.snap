// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
Array [
  Array [
    Object {
      "loader": "url-loader",
      "options": Object {
        "emitFile": false,
        "limit": 0,
        "name": "images/[name].[ext]",
        "outputPath": "static/chunks/media",
        "publicPath": "undefined_next/static/chunks/media",
      },
      "test": /\\\\\\.\\(png\\|jpg\\|jpeg\\|gif\\|webp\\|ico\\)\\$/,
    },
  ],
]
`;
